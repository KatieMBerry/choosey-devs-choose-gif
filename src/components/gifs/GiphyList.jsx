import React from 'react';
import PropTypes from 'prop-types';
import Giphy from './Giphy';
import styles from './GiphyList.module.css';

const GiphyList = ({ gifs }) => {
    const gifElements = gifs.map(gif => (
        <li key={gif.id}>
            <Giphy {...gif} />
        </li>
    ));

    return (
        <ul data-testid="gifs"
            className={styles.GiphyList}>
            {gifElements}
        </ul>
    );
};

GiphyList.propTypes = {
    gifs: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired
    }))
};

export default GiphyList;
