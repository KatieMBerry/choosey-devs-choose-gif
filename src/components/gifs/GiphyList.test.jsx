import React from 'react';
import { render, cleanup } from '@testing-library/react';
import GiphyList from './GiphyList';

describe('GiphyList component', () => {
    afterEach(() => cleanup());
    it('renders a list of GIFs', () => {
        const { asFragment } = render(<GiphyList
            gifs={[]}
        />);
        expect(asFragment()).toMatchSnapshot();
    });
});
