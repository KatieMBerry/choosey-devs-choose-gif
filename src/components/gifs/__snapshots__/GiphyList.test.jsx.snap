// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyList component renders a list of GIFs 1`] = `
<DocumentFragment>
  <ul
    class="undefined"
    data-testid="gifs"
  />
</DocumentFragment>
`;
